ethereum_history_api/oracles/src/script/prepareNoirFixtures.ts

```diff
@@ -43,6 +45,16 @@ for (const chain in FIXTURES) {
           await writeFile(join(modulePath, 'storage_proof.nr'), createStorageProofFixture(stateProof.storageProof));
           fixtureModules.push('storage_proof');
         }
+
+        if (transactionHashes) {
+          const transactionIndexes = transactionHashes.map((hash) => block.transactions.indexOf(hash));
+          const receiptProofs = await Promise.all(
+            transactionIndexes.map(async (txIdx) => await getReceiptProof(client, blockNumber, txIdx))
```

Comment:
Nope. This way you reconstruct the tree over and over for the same block. And do N http requests where N is the number of transactions within this fixture.
Let's refactor use the more low level function that getReceiptProof uses under the hood and avoid this promise.all.
1. Get all receipts.
2. Construct the tree.
3. Generate proofs from this tree.

Reply:
The transactionHashes array will always have length == 1. I will change the code so that we only take one transaction hash instead of an array.